src/Type/Php/StrContainingTypeSpecifyingExtension.php

```diff
@@ -76,21 +75,23 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 					$accessories[] = new AccessoryNumericStringType();
 				}
 
+				$rootExpr = $evaluatesToBoolean
+					? new BooleanAnd(
+						new NotIdentical(
+							$args[$needleArg]->value,
+							new String_(''),
+						),
+						new FuncCall(new Name('FAUX_FUNCTION_' . $functionReflection->getName()), $args),
+					)
+					: new FuncCall(new Name('FAUX_FUNCTION_' . $functionReflection->getName()), $args);
```

Comment:
This is a bit weird but needed to avoid false-positives for functions not evaluating to a boolean like `strpos`.

Reply:
Kept thinking about this. I think this cannot be there. And to get rid of it (and the error without it, see https://github.com/phpstan/phpstan-src/runs/7373184847?check_suite_focus=true) I'd need to check if the real expression (the function/method call) returns a boolean. For other functions that don't, like `strpos` the root expression should not be further specified. But to do that I'd need to check the return type via reflection I guess and this feels like too much here.

I'm not sure if this PR is the right approach. Maybe root expression specification should only be done where the FAUX_FUNCTION is created (-> the extension).